hipify-clang/src/HipifyAction.cpp

```diff
@@ -157,34 +157,40 @@ void HipifyAction::InclusionDirective(clang::SourceLocation hash_loc,
     }
 
     // Special-casing to avoid duplication of the hip_runtime include.
+    bool secondMainInclude = false;
     if (found->second.hipName == "hip/hip_runtime.h") {
         if (insertedRuntimeHeader) {
-            return;
+          secondMainInclude = true;
         }
-
         insertedRuntimeHeader = true;
     }
 
     Statistics::current().incrementCounter(found->second, file_name.str());
 
     clang::SourceLocation sl = filename_range.getBegin();
     if (found->second.unsupported) {
-        // An unsupported CUDA header? Oh dear. Print a warning.
         clang::DiagnosticsEngine& DE = getCompilerInstance().getDiagnostics();
         DE.Report(sl, DE.getCustomDiagID(clang::DiagnosticsEngine::Warning, "Unsupported CUDA header"));
         return;
     }
 
-    const char *B = SM.getCharacterData(sl);
+    char *B = nullptr;
```

Comment:
you can move this `const char *B = SM.getCharacterData(sl);` next to `if` statement and avoid `const_cast` below (by removing both `B` assigns in lines 183 and 192)